src/Umbraco.Cms.ManagementApi/Factories/ExamineIndexViewModelFactory.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Examine;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Infrastructure.Examine;
+using Umbraco.Cms.ManagementApi.ViewModels.ExamineManagement;
+
+namespace Umbraco.Cms.ManagementApi.Factories;
+
+public class ExamineIndexViewModelFactory : IExamineIndexViewModelFactory
+{
+    private readonly IIndexDiagnosticsFactory _indexDiagnosticsFactory;
+    private readonly IIndexRebuilder _indexRebuilder;
+
+    public ExamineIndexViewModelFactory(IIndexDiagnosticsFactory indexDiagnosticsFactory, IIndexRebuilder indexRebuilder)
+    {
+        _indexDiagnosticsFactory = indexDiagnosticsFactory;
+        _indexRebuilder = indexRebuilder;
+    }
+
+    public ExamineIndexViewModel Create(IIndex index)
+    {
+        var indexName = index.Name;
+
+        IIndexDiagnostics indexDiag = _indexDiagnosticsFactory.Create(index);
+
+        Attempt<string?> isHealthy = indexDiag.IsHealthy();
+
+        var properties = new Dictionary<string, object?>
```

Comment:
Is there a specific reason why we need these two properties in a generic dictionary instead of declaring them as explicit properties on `ExamineIndexViewModel`?